qiskit/_jobprocessor.py

```diff
@@ -0,0 +1,371 @@
+        if preformatted:
+            self.qobj = circuits
+            self.backend = self.qobj['config']['backend']
+            self.resources = {'max_credits':
+                              self.qobj['config']['max_credits']}
+        else:
+            self.backend = backend
+            self.resources = resources
+            # local and remote backends currently need different
+            # compilied circuit formats
+            formatted_circuits = []
+            if doCompile:
+                for circuit in self.circuits:
+                    formatted_circuits.append(None)
+            else:
+                if backend in self._local_backends:
+                    for circuit in self.circuits:
+                        formatted_circuits.append(openquantumcompiler.dag2json(circuit))
+                else:
+                    for circuit in self.circuits:
+                        formatted_circuits.append(circuit.qasm(qeflag=True))
+            # create circuit component of qobj
+            circuitRecords = []
+            if circuit_configs is None:
+                config = {'coupling_map': None,
+                          'basis_gates': 'u1,u2,u3,cx,id',
+                          'layout': None,
+                          'seed': seed}
+                circuit_configs = [config] * len(self.circuits)
+            for circuit, fcircuit, name, config in zip(self.circuits,
+                                                       formatted_circuits,
+                                                       self.names,
+                                                       circuit_configs):
+                record = {
+                    'name': name,
+                    'compiled_circuit': None if doCompile else fcircuit,
+                    'compiled_circuit_qasm': None if doCompile else fcircuit,
+                    'circuit': circuit,
+                    'config': config
+                }
+                circuitRecords.append(record)
+            qobjid = ''.join([random.choice(
+                string.ascii_letters + string.digits) for i in range(10)])
+            self.qobj = {'id': qobjid,
+                         'config': {
+                             'max_credits': resources['max_credits'],
+                             'shots': shots,
+                             'backend': backend
+                         },
+                         'circuits': circuitRecords
+            }
+        self.seed = seed
+        self.result = None
+        self.doCompile = doCompile
+
+
+class JobProcessor():
+    """
+    process a bunch of jobs and collect the results
+    """
+    def __init__(self, q_jobs, callback, max_workers=1, token=None, url=None, api=None):
+        """
+        Args:
+            q_jobs (QuantumJob | QuantumJob list | qobj): QuantumJob, list of
+                QuantumJob objects, or qobj dictionary.
+            online (bool): If true, establishes a connection to online server. This
```

Comment:
Another minor nitpick: could you drop `online` from the the docstring (and introduce `callback`) to match the signature?